platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/reconnect/StateBuilder.java

```diff
@@ -0,0 +1,100 @@
+     * It then uses the provided probabilities to modify the teacher state in order to emulate a scenario
+     * where the learner has disconnected from the network and hasn't updated its map with the latest changes:
+     * <ul>
+     *     <li>teacherAddProbability - a new node is added to the teacher state
+     *     <li>teacherRemoveProbability - an existing node is removed from the teacher state
+     *     <li>teacherModifyProbability - an existing node is updated with a new value in the teacher state
+     * </ul>
+     * <p>
+     * Note that the state populators must correctly support additions, updates, and removals (when the value is null).
+     *
+     * @param random a Random instance
+     * @param size the number of nodes in the learner state.
+     *          The teacher may have a slightly different number of nodes depending on the probabilities below.
+     * @param teacherAddProbability the probability of a key to be added to the teacher state
+     * @param teacherRemoveProbability the probability of a key to be removed from the teacher state
+     * @param teacherModifyProbability the probability of a node under a given key in the teacher state
+     *          to have a value that is different from the value under the same key in the learner state.
+     * @param teacherPopulator a BiConsumer that persists the teacher state (Map::put or similar)
+     * @param learnerPopulator a BiConsumer that persists the learner state (Map::put or similar)
+     */
+    public void buildState(
+            final Random random,
+            final long size,
+            final double teacherAddProbability,
+            final double teacherRemoveProbability,
+            final double teacherModifyProbability,
+            final BiConsumer<K, V> teacherPopulator,
+            final BiConsumer<K, V> learnerPopulator) {
+        LongStream.range(1, size).forEach(i -> {
+            final K key = keyBuilder.apply(i);
+            final V value = valueBuilder.apply(i);
+            teacherPopulator.accept(key, value);
+            learnerPopulator.accept(key, value);
+        });
+
+        LongStream.range(1, size).forEach(i -> {
+            // Make all random outcomes independent of each other:
+            final boolean teacherAdd = isRandomOutcome(random, teacherAddProbability);
+            final boolean teacherModify = isRandomOutcome(random, teacherModifyProbability);
+            final boolean teacherRemove = isRandomOutcome(random, teacherRemoveProbability);
+
+            if (teacherAdd) {
+                final K key = keyBuilder.apply(i + size);
+                final V value = valueBuilder.apply(i + size);
+                teacherPopulator.accept(key, value);
+            }
+
+            // Don't bother modifying if we're about to remove it
+            if (teacherRemove) {
+                final K key = keyBuilder.apply(i);
```

Comment:
It means, only original nodes (in range 1..size) may be removed from the map. It works just fine, but it would be even better to track the number of added nodes (in the teacherAdd branch above) and select a node to remove from range 1..size+added